CJSpohn/whats-cooking-refactor
# Feature/missing functionality

Is this a fix or a feature?
Feature

What is the change/fix?
Added a ton of functionality we're missing and refactored domUpdates to be more dynamic in all cases

Where should the reviewer start?
All changes are in domUpdates.js aside from one event listener in scripts

How should this be tested?
Check that there are no bugs when navigating pages, favoriting/unfavoriting, cookbooking/un cookbooking, and using the search bar to filter recipes by name.

Screenshots(if appropriate)
